modules/ugdk-core/include/ugdk/action/animationplayer.h

```diff
@@ -31,7 +31,7 @@ class AnimationPlayer : public MediaPlayer {
         frame_change_callback_ = callback;
     }
 
-    const Animation* current_animation() const {
+    std::shared_ptr<const Animation> current_animation() const {
```

Comment:
This is incorrect, but it's kinda IndexableTable's fault of returning raw pointers. However, I believe the table should own the pointers, and not return shared_ptrs. This means there's no STL pointer wrapper that fits the case.